### PR TITLE
Add Arc cannon

### PR DESCRIPTION
**Pull request by Masterliberty**
Arc Cannon with model, animations, tf1 weapon stats, and bug fixed _arc_cannon file.
Screenshot (2877)

### CODE REVIEW

**Review by GeckoEidechse — CHANGES_REQUESTED** [review]
Current version is missing weapon precaching

See https://github.com/R2Northstar/NorthstarMods/pull/276/files#diff-a449854f2677ae627216b17300ddcde22f3e3788d101a84299689c207cd1cefcR7 for reference on how to add it ^^

**Comment by GalacticMoblin** [comment]
I would just like to mention I have a custom made Arc Cannon weapon icon I made for Archon.

**Comment by ASpoonPlaysGames** [comment]
> I would just like to mention I have a custom made Arc Cannon weapon icon I made for Archon.

probably wont get used (yet) because uimg limit

**Review by GeckoEidechse — COMMENTED** [review]
Some quick one-player testing. Still need to test with other players:

- Arc Cannon is added to the game as intended
- Sound effects appear to still be missing. Separate PR?
- This will break compat both ways between combinations of server/client old/new. There's not a really a fix for this. We will have to version gate it when we release it either way.

To test are still
- [ ] Check the damage effect when getting hit by Arc Cannon
- [x] Have player without Arc Cannon join server with Arc Cannon (I expect them to be unable to connect as mentioned above)

**Comment by F1F7Y** [comment]
>  Sound effects appear to still be missing. Separate PR?

We can't add new sound effects so just like the triple threat itll have to stay silent

**Review by F1F7Y — COMMENTED** [review]
From skimming over the code it looks good. Due to this being based on r1 the code quality wont be up to r2 standards, but just like the triple threat this is ok as the game wont be updated with a newer squirrel compiler.

**Review by F1F7Y — APPROVED** [review]
Tested together with #660 on Launcher [#504](https://github.com/R2Northstar/NorthstarLauncher/pull/504)

Works

File: Northstar.Custom/mod/scripts/vscripts/weapons/mp_titanweapon_arc_cannon.nut
Comment (GeckoEidechse) [inline Northstar.Custom/mod/scripts/vscripts/weapons/mp_titanweapon_arc_cannon.nut:1]: Might commit arc cannon scripts separately first

**Review by GeckoEidechse — APPROVED** [review]
Works in testing. Of note is that this will break compatibility in the following way:

works: ✅
breaks: ❌

|                   | Server old | Server Arc Cannon |
|-------------------|------------|-------------------|
| Client old        | ✅        | ❌                |
| Cleint Arc Cannon | ✅        | ✅                |

**Comment by GeckoEidechse** [comment]
Alright Imma commit the original TF1 files first, then resolve merge conflicts myself and update the PR, then merge

File: Northstar.Custom/mod/scripts/vscripts/weapons/mp_titanweapon_arc_cannon.nut
Comment (GeckoEidechse) [reply Northstar.Custom/mod/scripts/vscripts/weapons/mp_titanweapon_arc_cannon.nut:1]: Done

**Pull request merged**